src/Draco.Chr/Tracing/MemoryTracer.cs

```diff
@@ -0,0 +1,47 @@
+using System.Collections.Generic;
+using System.IO;
+using Draco.Chr.Constraints;
+using Draco.Chr.Rules;
+
+namespace Draco.Chr.Tracing;
+
+/// <summary>
+/// A tracer logging into memory.
+/// </summary>
+public sealed class MemoryTracer : ITracer
+{
+    /// <summary>
+    /// The output stream.
+    /// </summary>
+    public MemoryStream Output { get; } = new();
```

Comment:
Why writing in a MemoryStream instead of shoving the trace objects in a list ?

Reply:
Renamed to TextTracer, as the purpose of this tracer was just to provide a simple text repr without needing to parameterize StreamTracer